### ISSUE TITLE
Variables not recognized.  Unknown token: 8:22 IDENT var.<VARNAME>

### ISSUE THREAD

**Issue by chumaumenze** [hashicorp/nomad#12390]
### Nomad version
Nomad v1.2.6 (a6c6b475db5073e33885377b4a5c733e1161020c)

### Operating system and Environment details
Tested on:
- macOS 12.2.1
- Ubuntu Focal (Multipass)

### Issue

```hcl
# /etc/nomad.d/server.hcl

variables {
  SERVER_LIST = [ "127.0.0.1" ]
  BOOTSTRAP_EXPECT = 1
}

server {
  enabled          = true
  bootstrap_expect = var.BOOTSTRAP_EXPECT

  server_join {
    retry_join = var.SERVER_LIST
    retry_max = 3
    retry_interval = "15s"
  }
}
```

Running

```sh
/usr/bin/nomad agent -config /etc/nomad.d/server.hcl
==> Error loading configuration from /etc/nomad.d/server.hcl: Error loading /etc/nomad.d/server.hcl: At 8:22: Unknown token: 8:22 IDENT var.BOOTSTRAP_EXPECT
```

### Reproduction steps

- Install the Nomad version (defined above)
- Create the server HCL config file
- Run the nomad agent using the config 

#### Expected Result

I expected the `var` to be parsed and not throw an error.

#### Actual Result

```sh
==> Error loading configuration from /etc/nomad.d/server.hcl: Error loading /etc/nomad.d/server.hcl: At 8:22: Unknown token: 8:22 IDENT var.BOOTSTRAP_EXPECT
```

**Comment by tgross** [comment]
Hi @chumaumenze! The `variables` block is for HCL2, which isn't currently supported at all in the agent configuration (this is implied by the HCL2 section of the docs being nestled under "job spec" but maybe that could be clearer).

We didn't implement HCL2 for configuration because running the agent is usually done by something like systemd, where there's no human being in the loop to add `-var` and `-var-file` flags. I'm going to mark this issue as a feature request (and change the title to match). But it would help us prioritize if you could describe a particular use case you had in mind.

**Comment by valodzka** [comment]
> We didn't implement HCL2 for configuration because running the agent is usually done by something like systemd, where there's no human being in the loop to add -var and -var-file flags.

Yes, but it can be used for some dynamic settings using environment variables (like `NOMAD_VAR_node_name=foo`) or as described in this (incorrectly closed) issue https://github.com/hashicorp/nomad/issues/1870
